### ISSUE TITLE
`reportUnknownVariableType` shouldn't be reported on assignments to `_`

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#686]
```py
_ = do_thing()
```
```
basedpyright: Type of "_" is partially unknown
  Type of "_" is "dict[Unknown, Unknown]" [reportUnknownVariableType]
```

depends on #140
